configure.ac

```diff
@@ -1283,6 +1283,107 @@ else
+                          yes/all  - all of the above features are enabled
+                          no/none  - none of the above features are enabled],,
+[enable_izem=no])
+
+# strip off multiple options, separated by commas
+save_IFS="$IFS"
+IFS=","
+for option in $enable_izem ; do
+    case "$option" in
+        sync)
+            izem_sync=yes
+        ;;
+        yes|all)
+            izem_sync=yes
+        ;;
+        no|none)
+        ;;
+        *)
+            IFS=$save_IFS
+            AC_MSG_WARN([Unknown value $option for enable-debug])
+            IFS=","
+        ;;
+    esac
+done
+IFS="$save_IFS"
+
+if test "$izem_sync" = "yes" ; then
+    AC_DEFINE(ENABLE_IZEM_SYNC,1,[Define to enable using Izem locks and condition variables])
+fi
+
+AC_ARG_VAR([ZMLIBNAME],[can be used to override the name of the Izem library (default: "zm")])
+ZMLIBNAME=${ZMLIBNAME:-"zm"}
+export ZMLIBNAME
+AC_SUBST(ZMLIBNAME)
+AC_ARG_WITH([zm-prefix],
+            [AS_HELP_STRING([[--with-zm-prefix[=DIR]]],
+                            [use the Izem library installed in DIR,
+                             rather than the one included in src/izem.  Pass
+                             "embedded" to force usage of the Izem source
+                             distributed with MPICH.])],
+            [],
+            [# see if Izem is already installed on the system
+             PAC_PUSH_FLAG([LIBS])
+             PAC_PREPEND_FLAG([-l${ZMLIBNAME}],[LIBS])
+             AC_LINK_IFELSE([AC_LANG_PROGRAM([dnl
+#include "lock/zm_ticket.h"
+],[
+zm_ticket_t lock;
+zm_ticket_init(&lock);
+zm_ticket_acquire(&lock);
+zm_ticket_release(&lock);
+])dnl
+                             ],
+                            [with_zm_prefix=system],[with_zm_prefix=embedded])
+             PAC_POP_FLAG([LIBS])
+             ])
+
+zmsrcdir=""
+AC_SUBST([zmsrcdir])
+zmlibdir=""
+AC_SUBST([zmlibdir])
+zmlib=""
+AC_SUBST([zmlib])
+
+if test "$enable_izem" != "no"; then
+    if test "$with_zm_prefix" = "embedded" ; then
+        if test -e "${use_top_srcdir}/src/izem" ; then
+            # no need for libtool versioning when embedding Izem
+            zm_subdir_args="--enable-embedded"
+            PAC_CONFIG_SUBDIR_ARGS([src/izem],[$zm_subdir_args],[],[AC_MSG_ERROR(Izem configure failed)])
+            zmsrcdir="${master_top_builddir}/src/izem"
+            zmlib="${master_top_builddir}/src/izem/src/lib${ZMLIBNAME}.la"
+            PAC_APPEND_FLAG([-I${use_top_srcdir}/src/izem/src/include],[CPPFLAGS])
+            PAC_APPEND_FLAG([-I${master_top_builddir}/src/izem/src/include],[CPPFLAGS])
+
+            PAC_CONFIG_SUBDIR_ARGS([src/izem],[$zm_subdir_args],[],[AC_MSG_ERROR([Izem configure failed])])
+        else
+            AC_MSG_WARN([Attempted to use the embedded Izem source tree in "src/izem", but it is missing.  Configuration or compilation may fail later.])
+        fi
+    elif test "$with_zm_prefix" = "system" ; then
+        PAC_PREPEND_FLAG([-l${ZMLIBNAME}],[WRAPPER_LIBS])
```

Comment:
Can you move the above `AC_LINK_IFELSE` code here to sanity check the installation?